librz/bin/p/bin_pe.inc

```diff
@@ -464,37 +451,130 @@ static RzPVector /*<RzBinImport *>*/ *imports(RzBinFile *bf) {
+		case PE_IMAGE_FILE_MACHINE_RISCV64:
+		case PE_IMAGE_FILE_MACHINE_RISCV128:
+			return "IMAGE_REL_BASED_RISCV_HIGH20";
+		default:
+			RZ_LOG_WARN("Unknown machine type for PE relocation type number: %d\n", reloc_type);
+			return "IMAGE_REL_BASED_UNKNOWN";
+		}
+
+	case PE_IMAGE_REL_BASED_THUMB_MOV32_RISCV_LOW12I:
+		switch (machine_type) {
+		case PE_IMAGE_FILE_MACHINE_THUMB:
+			return "IMAGE_REL_BASED_THUMB_MOV32";
+		case PE_IMAGE_FILE_MACHINE_RISCV32:
+		case PE_IMAGE_FILE_MACHINE_RISCV64:
+		case PE_IMAGE_FILE_MACHINE_RISCV128:
+			return "IMAGE_REL_BASED_RISCV_LOW12I";
+		default:
+			RZ_LOG_WARN("Unknown machine type for PE relocation type number: %d\n", reloc_type);
+			return "IMAGE_REL_BASED_UNKNOWN";
+		}
+	case PE_IMAGE_REL_BASED_RISCV_LOW12S_LOONGARCH32_MARK_LA:
+		switch (machine_type) {
+		case PE_IMAGE_FILE_MACHINE_RISCV32:
+		case PE_IMAGE_FILE_MACHINE_RISCV64:
+		case PE_IMAGE_FILE_MACHINE_RISCV128:
+			return "IMAGE_REL_BASED_RISCV_LOW12S";
+		case PE_IMAGE_FILE_MACHINE_LOONGARCH32:
+		case PE_IMAGE_FILE_MACHINE_LOONGARCH64:
+			return "IMAGE_REL_BASED_LOONGARCH32_MARK_LA";
+		default:
+			RZ_LOG_WARN("Unknown machine type for PE relocation type number: %d\n", reloc_type);
+			return "IMAGE_REL_BASED_UNKNOWN";
+		}
+	case PE_IMAGE_REL_BASED_MIPS_JMPADDR16:
+		switch (machine_type) {
+		case PE_IMAGE_FILE_MACHINE_MIPS16:
+		case PE_IMAGE_FILE_MACHINE_MIPSFPU:
+		case PE_IMAGE_FILE_MACHINE_MIPSFPU16:
+		case PE_IMAGE_FILE_MACHINE_R4000:
+		case PE_IMAGE_FILE_MACHINE_WCEMIPSV2:
+			return "IMAGE_REL_BASED_MIPS_JMPADDR16";
+		default:
+			RZ_LOG_WARN("Unknown machine type for PE relocation type number: %d\n", reloc_type);
+			return "IMAGE_REL_BASED_UNKNOWN";
+		}
+	case PE_IMAGE_REL_BASED_DIR64:
+		return "IMAGE_REL_BASED_DIR64";
+	default:
+		RZ_LOG_WARN("Unknown PE relocation type number: %d\n", reloc_type);
+		return "IMAGE_REL_BASED_UNKNOWN";
+	}
+}
+
 static RzPVector /*<RzBinReloc *>*/ *relocs(RzBinFile *bf) {
-	struct PE_(rz_bin_pe_obj_t) *obj = bf->o->bin_obj;
-	if (obj) {
-		return obj->relocs;
+	// NOTE: No patching is needed because base relocations (as their name suggests) patch relocations based off the delta
+	// between the executable expected load address, and the actual load address. And since Rizin always loads
+	// the binaries at their preferred base address, the delta is always 0 and so no patching is needed.
+	rz_return_val_if_fail(bf && bf->o && bf->o->bin_obj, NULL);
+	RzBinPEObj *bin = bf->o->bin_obj;
+
+	if (!bin->relocs) {
+		return NULL;
 	}
-	return NULL;
+
+	RzPVector *ret = NULL;
+	if (!(ret = rz_pvector_new((RzPVectorFree)rz_bin_reloc_free))) {
+		return NULL;
+	}
+
+	if (!PE_(bin_pe_has_base_relocs)(bin)) {
+		return NULL;
+	}
+
+	RzBinPeRelocEnt *pe_reloc;
+	rz_vector_foreach (bin->relocs, pe_reloc) {
+		RzBinReloc *general_reloc = RZ_NEW0(RzBinReloc);
+		if (!general_reloc) {
+			RZ_LOG_WARN("Failed to convert PE reloc to general reloc\n");
+			rz_bin_reloc_free(general_reloc);
+			continue;
```

Comment:
if alloc fails, then there is no memory left.
You should probably do the following instead of calling `continue`:

```
rz_vector_free(bin->relocs);
return ret;
```